### PR TITLE
 emb6: allow for warnings to be disabled

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
Currently, due to the way emb6 is compiled in our build system, it is
not possible to deactivate certain warnings. This fix provides a
solution for that, by building the submodules in the package build step
instead of on their own, similar to how JerryScript is build.

Additionally, I piggy-backed to commits concerning warnings:

1. emb6: ignore implicit fallthrough errors by GCC 7 (see #8430)
2. emb6: remove patch to fix warnings and ignore them instead, to make the package easier to maintain in the future (less patches lead to less conflicts when applying those patches)

### Issues/PRs references
See https://github.com/RIOT-OS/RIOT/pull/8430/files#r168480449

### CODE REVIEW

**Comment by smlng** [comment]
nice!

**Comment by miri64** [comment]
Build error looks unrelated...

File: pkg/emb6/Makefile.include [hunk 1]
```diff
@@ -4,6 +4,15 @@ EMB6_CONTRIB := $(RIOTBASE)/pkg/emb6/contrib
 
 INCLUDES += -I$(PKG_BUILDDIR)/target
 INCLUDES += -I$(RIOTBASE)/pkg/emb6/include
+INCLUDES += -I$(EMB6_DIR)/emb6
```
Comment (kaspar030) [inline pkg/emb6/Makefile.include]: are these globally needed?

**Comment by smlng** [comment]
> Build error looks unrelated...

yes it is, I was testing some new CI workers (with new packages versions) - didn't work out (yet), removed them again. I'll retrigger murdock ...

File: pkg/emb6/Makefile.include (hunk 1 above)
Comment (miri64) [reply pkg/emb6/Makefile.include]: Yes, either the application or `auto_init` (if we ever do that for `emb6`) need the emb6 headers to bootstrap the stack.

**Pull request merged**